plaso/output/elastic.py

```diff
@@ -13,7 +13,70 @@ class ElasticsearchOutputModule(shared_elastic.SharedElasticsearchOutputModule):
 
   NAME = 'elastic'
   DESCRIPTION = 'Saves the events into an Elasticsearch database.'
-
+  DEFAULT_MAPPING = {
+      'properties': {
+          "application": {
+              "type": "text",
+              "fields": {
+                  "keyword": {"type": "keyword"}},
+          },
+          "data": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "doc_type": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "event_type": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "exit_status": {"type": "text",
+                          "fields": {"keyword": {"type": "keyword"}},
+                         },
+          "facility": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "file_reference": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "file_size": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "flags": {"type": "text", "fields": {"keyword": {"type": "keyword"}}},
+          "identifier": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "message_status": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "message_type": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "offset": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "sequence_number": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "severity": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "source_port": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "user_identifier": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "version": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
```

Comment:
since the mapping is more or less the same, I'm wondering whether we should have this DEFAULT_MAPPING simpler and then to generate the actual mapping.
```
DEFAULT_MAPPING = {
   'version': 'text',
   'user_identifier': 'text',
...
}
```

Then in the mappings we can do

```
mappings = {}
for key, type_string in self.DEFAULT_MAPPING.items():
  mappings[key] = {
      'type': type_string, 'fields': {'keyword': {'type': 'keyword'}
  }
```

something like that, then the original dict is more easily readable, and perhaps easier to maintain

Reply:
I don't think that generate the mapping from an alternative notation would make it easier to maintain.
The advantage of the current solution is that it fits the elasticsearch syntax. Using an intermediate notation would make it harder to make the link between the code and the mapping that will be applied.

Furthermore, it could be interesting in the future to create a more specific mapping (e.g: specific tokenizer for path, registry key, (https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-pathhierarchy-tokenizer.html), store the length of some filename, etc...). That would allow more powerful/faster query.

(Btw, currently it easy to make it by creating your own index before upload with psort)
Keep the elasticsearch syntax would also be more friendly for someone that want to extend the mapping with his own customization.

Reply:
ACK and yes I agree, we should start looking more into different type of tokenization for these, the same goes for dfdatetime objects later in the future, that may represent `fuzzy` timestamps (as in will contain a timestamp and an accuracy, essentially making it a range) or ranges.